interfaces/systemd/backend.go

```diff
@@ -69,16 +74,22 @@ func (b *Backend) Setup(snapInfo *snap.Info, confinement interfaces.ConfinementO
 	}
 	glob := interfaces.InterfaceServiceName(snapName, "*")
 
-	systemd := sysd.New(dirs.GlobalRootDir, sysd.SystemMode, &dummyReporter{})
+	var systemd sysd.Systemd
+	if b.preseed {
+		systemd = sysd.NewEmulationMode(dirs.GlobalRootDir)
```

Comment:
Do we need something that is both emulation and system mode?

Reply:
This how emulation mode works. It errors on things it can't do, they could be no-op instead, I think that would work ok with out current usages, but it could hide problems in some other cases, and it's not the path preseed followed so far. We might reconsider this but this PR is probably not the place or time for that.